app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  skip_before_action :verify_authenticity_token
+  
+  def new
+    build_resource({})
+    resource.build_profile
+    respond_with resource
+  end
+
+  def create
+    build_resource(sign_up_params)
+
+    if resource.valid? && resource.save
```

Comment:
I think that `resource.save` will return false if it's not valid, so you can remove `resource.valid? &&`. I'm not 100% sure, so you might want to double check before changing it